src/value.rs

```diff
@@ -90,6 +99,50 @@ impl AllocatedValue {
     }
 }
 
+impl SignedInteger {
+    pub fn zero() -> Self {
+        SignedInteger::Positive(0)
+    }
+
+    pub fn abs(&self) -> u64 {
+        match self {
+            SignedInteger::Positive(x) => *x,
+            SignedInteger::Negative(x) => *x,
+        }
+    }
+
+    pub fn sign(&self) -> u8 {
+        match self {
+            SignedInteger::Positive(_) => 1,
+            SignedInteger::Negative(_) => 0,
+        }
+    }
+}
+
+impl From<u64> for SignedInteger {
+    fn from(u: u64) -> SignedInteger {
+        SignedInteger::Positive(u)
+    }
+}
+
+impl Into<Scalar> for SignedInteger {
+    fn into(self) -> Scalar {
+        unimplemented!()
+    }
+}
+
+impl ConditionallySelectable for SignedInteger {
+    fn conditional_select(a: &Self, b: &Self, choice: Choice) -> Self {
+        let val = u64::conditional_select(&a.abs(), &b.abs(), choice);
+        let sign = u8::conditional_select(&a.sign(), &b.sign(), choice);
+        if sign == 0 {
```

Comment:
This is not a const-time operation, but we can convert SignedInteger to `i128` and then back from it.

Reply:
and perform `conditional_select` on i128.

Reply:
Although, i admit this sign()/abs() stuff is cool.